File: internal/common/models/order.go
package models

// OrderStatus mirrors the TS enum used in the Node project.
type OrderStatus string

const (
	OrderStatusCreated   OrderStatus = "created"
	OrderStatusCancelled OrderStatus = "cancelled"
	OrderStatusAwaiting  OrderStatus = "awaiting:payment"
	OrderStatusComplete  OrderStatus = "complete"
)

type Order struct {
	ID       string      `json:"id"`
	UserID   string      `json:"userId"`
	Status   OrderStatus `json:"status"`
	TicketID string      `json:"ticketId"`
	Version  int         `json:"version"`
}

File: frontend/src/lib/types/order.ts
export enum OrderStatus {
  Created = 'created',
  Cancelled = 'cancelled',
  Complete = 'complete',
}

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: string;
  ticket: {
    id: string;
    title: string;
    price: number;
  };
  version: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrderInput {
  ticketId: string;
}
